curl-helper.c

```diff
@@ -865,6 +867,48 @@ static size_t cb_WRITEFUNCTION2(char *ptr, size_t size, size_t nmemb, void *data
     return r;
 }
 
+static size_t cb_WRITEFUNCTION_BUF(char *ptr, size_t size, size_t nmemb, void *data)
+{
+  caml_leave_blocking_section();
+
+  CAMLparam0();
+  CAMLlocal2(result, buf);
+  Connection *conn = (Connection *)data;
+
+  checkConnection(conn);
+  buf = Field(conn->ocamlValues, Ocaml_CALLBACKBUFFER);
+  if (buf == Val_unit)
+    {
+      buf = caml_ba_alloc_dims(CAML_BA_UINT8 | CAML_BA_C_LAYOUT | CAML_BA_EXTERNAL, 1, NULL, 0);
```

Comment:
i actually had in mind allocating when set_writefunction_buf is called but this way also works indeed.